sdcm/utils/cluster_tools.py

```diff
@@ -10,12 +10,61 @@
 # See LICENSE for more details.
 #
 # Copyright (c) 2025 ScyllaDB
+from __future__ import annotations
+
+import logging
+from typing import TYPE_CHECKING
 from collections import defaultdict
 
+if TYPE_CHECKING:
+    from sdcm.cluster import BaseCluster, BaseNode
+
+LOGGER = logging.getLogger(__name__)
 
-def group_nodes_by_dc_idx(nodes: list['BaseNode']) -> dict[int, list['BaseNode']]:  # noqa: F821
+
+def group_nodes_by_dc_idx(nodes: list[BaseNode]) -> dict[int, list[BaseNode]]:  # noqa: F821
     """ Group nodes by dc_idx """
     nodes_by_dc_idx = defaultdict(list)
     for node in nodes:
         nodes_by_dc_idx[node.dc_idx].append(node)
     return nodes_by_dc_idx
+
+
+def is_cluster_rack_balanced(db_cluster: BaseCluster) -> bool:  # noqa: F821
+    """ Check if the cluster is rack balanced, meaning that each rack has an equal number to other racks in the same datacenter"""
+    nodes_by_dc_idx = group_nodes_by_dc_idx(db_cluster.nodes)
+    rack_balancing_check_mode = db_cluster.params.get("rack_balancing_check_mode") or "naive"
+
+    db_nodes_config_count = [int(i) for i in str(db_cluster.params.get("n_db_nodes") or 0).split(" ")]
+
+    if rack_balancing_check_mode == "naive":
+        for dc_idx, nodes_in_dc in nodes_by_dc_idx.items():
+            racks = defaultdict(int)
+            for node in nodes_in_dc:
+                racks[node.rack] += 1
+            if len(set(racks.values())) != 1:
+                # print the rack distribution for debugging
+                LOGGER.debug(f"Datacenter {dc_idx} rack distribution: {dict(racks)}")
+                return False
+        return True
+    elif rack_balancing_check_mode == "per-initial_config":
+        for dc_idx, nodes_in_dc in nodes_by_dc_idx.items():
+            racks = defaultdict(int)
+            for node in nodes_in_dc:
+                racks[node.rack] += 1
+            # Check if the number of nodes in each rack matches the initial configuration
+            if len(racks) != db_cluster.racks_count:
+                LOGGER.debug(f"Datacenter {dc_idx} rack distribution: {dict(racks)}")
+                return False
+            # Check if all racks have the same number of nodes
+            if db_nodes_config_count[dc_idx] != len(nodes_in_dc):
+                LOGGER.debug(
+                    f"Datacenter {dc_idx} rack distribution: {dict(racks)}, config count: {db_nodes_config_count[dc_idx]}, {len(nodes_in_dc)=}")
+                return False
+        return True
+    elif rack_balancing_check_mode == "disabled":
+        # If rack balancing mode is 'none', we don't care about rack balancing
```

Comment:
in this case we could just verify if #nodes equals initial #, regardless of racks distribution. (anyway we should verify it always).
Can happen in situation when we decommissioned 3 nodes equally from racks and couldn't provision any due to capacity issue - racks would match and no error raised when `naive` config was used

Reply:
so how it's different from just stop the test on any case of capacity error 

the once case we can continue, if this nemesis added one node, and failed on capacity.

all case that we first decommission would end up not as configuration 

so the whole idea of catching the capactity errors and skip is quite sketchy

Reply:
change to have just the per configuration option